R/compare.R

```diff
@@ -314,6 +329,55 @@ expand_diff_text <- function(old, new) {
   list(old = old, new = new)
 }
 
+merge_packages <- function(old, new) {
+  opkgs <- parse_pkgs(old)
+  npkgs <- parse_pkgs(new)
+
+  if (is.null(opkgs) || is.null(opkgs)) return(list(old = old, new = new))
+
+  names_to_keep <- c("package", "version", "source")
+  opkgs$pkgs <- opkgs$pkgs[names_to_keep]
+  npkgs$pkgs <- npkgs$pkgs[names_to_keep]
+  names(opkgs$pkgs) <- c("package", "old_version", "old_source")
+  names(npkgs$pkgs) <- c("package", "new_version", "new_source")
+
+  merge_res <- merge(opkgs$pkgs, npkgs$pkgs, all = TRUE)
+  merge_res <- with(merge_res,
+       data.frame(
+         package = package,
+         "v!=" = mark(is_different(old_version, new_version)),
+         old_version = old_version,
+         new_version = new_version,
+         "s!=" = mark(is_different(old_source, new_source)),
+         old_source = old_source,
+         new_source = new_source,
```

Comment:
I definitely need to do some truncating here, in the case of a package installed from GitHub.